its/src/test/java/com/sonar/it/csharp/ExternalIssuesTest.java

```diff
@@ -0,0 +1,106 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.it.csharp;
+
+import com.sonar.it.shared.TestUtils;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sonarqube.ws.Issues;
+
+import static com.sonar.it.csharp.Tests.ORCHESTRATOR;
+import static com.sonar.it.csharp.Tests.getComponent;
+import static com.sonar.it.csharp.Tests.getIssues;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonarqube.ws.Common.RuleType;
+
+public class ExternalIssuesTest {
+
+  @Rule
+  public TemporaryFolder temp = TestUtils.createTempFolder();
+
+  private static final String PROJECT = "ExternalIssues";
+  private static final String PROGRAM_COMPONENT_ID = "ExternalIssues:Program.cs";
+  private static final String SONAR_RULES_PREFIX = "csharpsquid:";
+  // note that in the UI the prefix will be 'roslyn:'
+  private static final String ROSLYN_RULES_PREFIX = "external_roslyn:";
+
+
+  @Before
+  public void init() {
+    TestUtils.reset(ORCHESTRATOR);
+  }
+
+  @Test
+  public void external_issues_imported_by_default_as_code_smells() throws Exception {
+    Tests.analyzeProject(temp, PROJECT, null);
+
+    assertThat(getComponent(PROGRAM_COMPONENT_ID)).isNotNull();
+    List<Issues.Issue> allIssues = getIssues(PROGRAM_COMPONENT_ID);
+
+    assertThat(allIssues).hasSize(5);
+    assertThat(filter(allIssues, SONAR_RULES_PREFIX, RuleType.CODE_SMELL)).hasSize(1);
+    assertThat(filter(allIssues, ROSLYN_RULES_PREFIX, RuleType.CODE_SMELL)).hasSize(4);
+  }
+
+  @Test
+  public void external_issues_are_ignored() throws Exception {
+    Tests.analyzeProject(temp, PROJECT, null,
+      "sonar.cs.roslyn.ignoreIssues", "true");
+
+    assertThat(getComponent(PROGRAM_COMPONENT_ID)).isNotNull();
+    List<Issues.Issue> allIssues = getIssues(PROGRAM_COMPONENT_ID);
+
+    assertThat(allIssues).hasSize(1); // our rule
+    assertThat(filter(allIssues, ROSLYN_RULES_PREFIX)).isEmpty();
+  }
+
+  @Test
+  public void external_issues_categories_mapping() throws Exception {
+    Tests.analyzeProject(temp, PROJECT, null,
+      "sonar.cs.roslyn.bugCategories", "StyleCop.CSharp.DocumentationRules,StyleCop.CSharp.MaintainabilityRules",
+      "sonar.cs.roslyn.vulnerabilityCategories", "StyleCop.CSharp.OrderingRules");
+
+    assertThat(getComponent(PROGRAM_COMPONENT_ID)).isNotNull();
+    List<Issues.Issue> allIssues = getIssues(PROGRAM_COMPONENT_ID);
+
+    assertThat(filter(allIssues, ROSLYN_RULES_PREFIX, RuleType.CODE_SMELL)).isEmpty();
```

Comment:
Would be nice to add one more case to map one of the Roslyn categories to code smell to be able to cover also that (`sonar.cs.roslyn.codeSmellCategories`). 

Note: I like that we cover the cases with `no category mapped` and with `multiple categories mapped` and I would keep them.